File: codeforces/1400c.py
T = int(input())
for _ in range(T):
    S = input()
    N = int(input())
    L = len(S)
    arr = [0]*L
    iszero = [False]*L
    for i in range(L):
        if (S[i] == '0'):
            if (i-N >= 0): iszero[i-N] = True
            if (i+N < L): iszero[i+N] = True
    used = False
    for i in range(L):
        if (S[i] == '1'):
            used = False
            if (i-N>=0):
                if (not iszero[i-N]):
                    used = True
                    arr[i-N] = 1    
            if (i+N<L):
                if (not iszero[i+N]):
                    used = True
                    arr[i+N] = 1
            if (not used):
                print(-1)
                break
    else:
        for i in range(L):
            print(arr[i], end='')
        print('')

File: codeforces/1400c.cpp
#include <iostream>
#include <string>
using namespace std;

int arr[100001];
bool iszero[100001];
int main() {
    int T; cin>>T;
    while (T--) {
        string S; cin>>S;
        int N; cin>>N;
        int L = S.length();
        fill_n(arr, L+1, 0);
        fill_n(iszero, L+1, false);
        for (int i=0; i<L; i++) {
            if (S.at(i) == '0') {
                if (i-N >= 0) iszero[i-N] = true;
                if (i+N < L) iszero[i+N] = true;
            }
        }
        bool used;
        for (int i=0; i<L; i++) {
            if (S.at(i) == '1') {
                used = false;
                if (i-N>=0) {
                    if (!iszero[i-N]) {
                        used = true;
                        arr[i-N] = 1;    
                    }
                }
                if (i+N<L) {
                    if (!iszero[i+N]) {
                        used = true;
                        arr[i+N] = 1;
                    }
                }
                if (!used) {
                    cout<< -1 <<'\n';
                    break;
                }
            }
        }
        if (used) {
            for (int i=0; i<L; i++) {
                cout<<arr[i];
            }
            cout<<'\n';
        }
    }
}